wp/lib/bap_wp/src/precondition.ml

```diff
@@ -472,7 +472,35 @@ let spec_verifier_nondet (sub : Sub.t) (_ : Arch.t) : Env.fun_spec option =
                | None -> failwith "Verifier headerfile must be specified with --api-path" in
              let vars = output |> Bap.Std.Arg.rhs |> Exp.free_vars in
              let v = Var.Set.choose_exn vars in
-             subst_fun_outputs env sub post ~inputs:[] ~outputs:[v], env)
+             let z3_v, env = Env.get_var env v in
+             let name = Format.sprintf "%s_ret_%s" (Sub.name sub) (Expr.to_string z3_v) in
+             let fresh = new_z3_expr env ~name:name (Var.typ v) in
+             Constr.substitute_one post z3_v fresh, env)
+    }
+  else
+    None
+
+let spec_calloc (sub : Sub.t) (_ : Arch.t) : Env.fun_spec option =
+  if String.equal (Sub.name sub) "calloc" then
+    let open Env in
+    Some {
+      spec_name = "spec_calloc";
+      spec = Summary
+        (* Currently stubbing this spec to mirror that of spec_verifier_nondet. *)
```

Comment:
Are we getting code duplication from `spec_verifier_nondet`?

Reply:
That is true. I can add the calloc and malloc case to `nondet`'s spec.